YourEasyRent/Controllers/ProductController.cs

```diff
@@ -63,6 +63,10 @@ public async Task<ActionResult<IEnumerable<Product>>> GetProductByBrand(string b
             try
             {
                 var brandProducts = await _repository.GetByBrand(brand);
+                if (brandProducts == null)
```

Comment:
It is a "best practice" to never return NULL from methods that return a collection. You should just return an empty List.

And return an OK(200 statuscode) result with an empty list in it.

`NotFound` is mostly for wrong urls or single record fetches lik`GET products/{id}`

Reply:
Still aint fixed